src/litserve/loops.py

```diff
@@ -573,6 +575,239 @@ def __call__(
         )
 
 
+def notify_timed_out_requests(
+    response_queues: List[Queue],
+    timed_out_uids: List[Tuple[int, str]],
+):
+    for response_queue_id, uid in timed_out_uids:
+        logger.error(f"Request {uid} was waiting in the queue for too long and has been timed out.")
+        response_queues[response_queue_id].put((uid, (HTTPException(504, "Request timed out"), LitAPIStatus.ERROR)))
+
+
+@dataclass
+class Output:
+    """Outputs from a single step of the loop."""
+
+    uid: str
+    output: Any
+    status: LitAPIStatus
+
+
+class LitLoop(_BaseLoop):
+    def __init__(self):
+        self._context = {}
+
+    def get_batch_requests(self, lit_api: LitAPI, request_queue: Queue, max_batch_size: int, batch_timeout: float):
+        if max_batch_size <= 1:
+            raise ValueError("max_batch_size must be greater than 1")
```

Comment:
I think we should allow batch size 1 here, We may want to teak the batch size without opting for a different loop altogether

Reply:
we have [get_request](https://github.com/Lightning-AI/LitServe/pull/387/files#diff-09cc07bb152f3c2d8b85404f472e5f4516835d1236155b431f48690d09ee790bR612) method to get the non-batched request. 

Two different methods because we need to check for timeout and notify users during batching.